src/viking.js

```diff
@@ -1,17 +1,89 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(dmg) {
+    this.health -= dmg;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(dmg) {
+    this.health -= dmg;
+    return this.health > 0
+      ? `${this.name} has received ${dmg} points of damage`
+      : `${this.name} has died in act of combat`;
+  }
+  battleCry() {
+    return `Odin Owns You All!`;
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  receiveDamage(dmg) {
+    this.health -= dmg;
+    return this.health > 0
+      ? `A Saxon has received ${dmg} points of damage`
+      : `A Saxon has died in combat`;
+  }
+}
 
 // War
-class War {}
-
-
-
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+  addViking(newViking) {
+    this.vikingArmy.push(newViking);
+  }
+  addSaxon(newSaxon) {
+    this.saxonArmy.push(newSaxon);
+  }
+  vikingAttack() {
+    const randomizerSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+    const randomizerViking = Math.floor(Math.random() * this.vikingArmy.length);
+    const randomSaxon = this.saxonArmy[randomizerSaxon];
+    const randomViking = this.vikingArmy[randomizerViking];
```

Comment:
Here, if you wanted make a shorter solution, you could access the array instead of creating those variables:
```
const randomViking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)]
```